controllers/heat_controller.go

```diff
@@ -568,6 +568,22 @@ func (r *HeatReconciler) reconcileNormal(ctx context.Context, instance *heatv1be
 		return ctrlResult, err
 	}
 
+	// remove finalizers from previous MariaDBAccounts for which we have
+	// switched.
+	// TODO(zzzeek) - It's not clear if this is called too early here.
+	// at the moment, heat_controller_test.go doesn't seem to have fixtures
+	// I can use to simulate getting all the way to the end of a reconcile
+	// for an instance.  Basically this should be called when any pods have
+	// been restarted to run on an updated set of DB credentials, and the old
+	// ones are no longer needed.  This would allow the scenario where
+	// a new MariaDBAccount is created and an old MariaDBAccount is marked
+	// deleted at once, where the finalizer will keep the old one around until
+	// it's safe to drop.
+	err = mariadbv1.DeleteUnusedMariaDBAccountFinalizers(ctx, helper, instance.Name, instance.Spec.DatabaseAccount, instance.Namespace)
```

Comment:
so I have a suite of ginkgo tests that exercises the finalizers moving from one MariaDBAccount to another, once services are redeployed with an updated MariaDBAccount / username/password.    That's the suite thats added in the tests here.   It wont work if the DeleteUnusedMariaDBAccountFinalizers call above isn't called, since that's part of what it's testing.   

Right now, having the call where it is is *mostly* fine, you can rotate a new MariaDBAccount in and all will be fine.  The one thing you can't do here is delete the old MariaDBAccount before the new one is fully reconciled and running, without the risk of a brief race condition where existing pods are using the old database login that would be pulled out, since the finalizer here is removed slightly too early to be guaranteed of that.     The issue would of course be worked out within a few seconds in the average case and wouldn't be noticed, and we as yet dont have any system that is actually rotating out old/new MariaDBAccount objects, so there will be no impact from this for now.  However, the comment here should be kept that we would like to move this call downwards to be after we are ensured all pods have been re-deployed with new credentials.